syncthing-bep/src/main/kotlin/net/syncthing/java/bep/BlockPuller.kt

```diff
@@ -51,10 +53,12 @@ object BlockPuller {
         val (newFileInfo, fileBlocks) = indexHandler.getFileInfoAndBlocksByPath(fileInfo.folder, fileInfo.path) ?: throw FileNotFoundException()
 
         if (fileBlocks.hash != fileInfo.hash) {
-            throw IllegalStateException("the current file entry hash does not match the hash of the provided one")
+            val exception = IllegalStateException("The current file entry hash does not match the hash of the provided one.")
+            LOGGER.atInfo().withThrowable(exception).log("Expected hash: ( {} ), Provided hash: ( {} ).",fileInfo.hash,fileBlocks.hash)
```

Comment:
Don't include the exception, since it is being generated at this location. This should also be at least `warn` level. Use a message like `The current file entry hash ({}) does not match the provided one ({}).`. No spaces inside the parenthesis. In this case, since the exception is going to get thrown anyhow, I'd write it like,

```kotlin
val message = "The current file entry hash (${fileInfo.hash}) does not match the provided one (${fileBlocks.hash})."
val exception = IllegalStateException(message)
LOGGER.atError().log(message)
throw exception
```

Make sure to put spaces after your method argument commas.